addons/account_tax_python/models/account_tax.py

```diff
@@ -47,11 +48,19 @@ def compute_all(self, price_unit, currency=None, quantity=1.0, product=None, par
         for tax in self.filtered(lambda r: r.amount_type == 'code'):
             localdict = self._context.get('tax_computation_context', {})
             localdict.update({'price_unit': price_unit, 'quantity': quantity, 'product': product, 'partner': partner, 'company': company})
-            safe_eval(tax.python_applicable, localdict, mode="exec", nocopy=True)
+            tax.evaluate_expression_code(tax.python_applicable, localdict)
             if localdict.get('result', False):
                 taxes += tax
         return super(AccountTaxPython, taxes).compute_all(price_unit, currency, quantity, product, partner, is_refund=is_refund, handle_price_include=handle_price_include)
 
+    def evaluate_expression_code(self, code, localdict):
```

Comment:
This function should be private. 
And actually, even so, I'm pretty sure the security team won't like it. Indeed, your function could be used from anywhere within the code to call a safe_eval without triggering a security warning. We don't want that. 

Essentially, we're just trying to avoid duplicating a raise statement here. I thing we can simply generate the error message from a function, and keep two explicit safe_eval doing each an explicity try/except and raise. Once this is done, you can request a review from rd-security.

Reply:
Ok but why do we need a function to just raise an error message. Instead of that can't we simply raise the message in the try except itself? What is your suggestion? Let me know if I understood your message wrong way.

Reply:
Not a function raising, a function returning the string of the error message to be raised. Just to avoid duplicating it. But indeed, if you think it's too much, you can copy paste the same string twice, I don't mind ^^